### PR TITLE
strongswan: miscellaneous cleanup of config file, etc

### PR DESCRIPTION
**Pull request by pprindeville**
Maintainer: me, @Thermi
Compile tested: x86_64, generic, HEAD (db9784bedd)
Run tested: same, installed on production router

Description:

Forbid absolute/relative paths for certificate and key files.  Get rid of last remnants of left/right naming.  Factor invariant code paths.  Drop redundant `secrets.rsa.filename` section.

Also add a couple of additional new settings.

### CODE REVIEW

**Comment by pprindeville** [comment]
@CodeFetch Have a look.

**Comment by pprindeville** [comment]
Contains some of the fixes from PR #15922.

File: net/strongswan/files/swanctl.init [hunk 1]
```diff
@@ -299,12 +301,23 @@ config_child() {
 		;;
 	esac
 
+	case "$hw_offload" in
+	yes|auto|"")
```
Comment (Thermi) [inline net/strongswan/files/swanctl.init:307]: Add "no".

File: libs/libzip/Makefile [hunk 2]
```diff
@@ -6,12 +6,12 @@
 include $(TOPDIR)/rules.mk
```
Comment (Thermi) [inline libs/libzip/Makefile:6]: This shouldn't be in this PR at all.

File: net/strongswan/files/swanctl.init [hunk 3]
```diff
@@ -219,6 +219,7 @@ config_child() {
 	local mode="$2"
```
Comment (Thermi) [inline net/strongswan/files/swanctl.init:219]: The description of the commit is not correct.

The "connections.<conn>.children.<child>.interface" key in swanctl sets the interface of the policy. It does not influence routing.

This means that the policy will have the additional criteria of the outgoing interface being the value of this setting.
Means, you can potentially have different policies (and thus states) for different outgoing interfaces. That can enable routing of traffic going out of different interfaces to different peers.

**Review by Thermi — CHANGES_REQUESTED** [review]

File: net/strongswan/files/swanctl.init (hunk 1 above)
Comment (pprindeville) [reply net/strongswan/files/swanctl.init]: Fixed.

File: libs/libzip/Makefile (hunk 2 above)
Comment (pprindeville) [reply libs/libzip/Makefile]: Doh. Not sure how that got there.  Rebased.

File: net/strongswan/files/swanctl.init (hunk 3 above)
Comment (pprindeville) [reply net/strongswan/files/swanctl.init:219]: Fixed.

**Pull request merged**